app/src/main/java/com/globant/weatherapp/mvp/presenter/WeatherPresenter.kt

```diff
@@ -1,13 +1,27 @@
 package com.globant.weatherapp.mvp.presenter
 
 import com.globant.weatherapp.mvp.contracts.WeatherContracts
+import com.globant.weatherapp.services.WeatherService
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
 
 class WeatherPresenter(
     private val model: WeatherContracts.Model,
-    private val view: WeatherContracts.View
+    private val view: WeatherContracts.View,
+    private val service: WeatherService = WeatherService()
 ) : WeatherContracts.Presenter {
 
-    override fun initPresenter() {
+    override fun initPresenter(cityId: Int) {
+        getFiveDaysWeather(cityId)
         view.initView()
     }
+
+    fun getFiveDaysWeather(cityId: Int) {
+        service.getFiveDaysWeatherByCityId(cityId)
```

Comment:
the presenter do not know the service.. that is the responsibility of the model.
You should call here model.getFiveDaysWeatherByCityId(cityId) and the model will call service.getFiveDaysWeatherByCityId(cityId).

Reply:
Okey, done!